libcontainer/utils/utils_unix.go

```diff
@@ -146,45 +144,13 @@ func NewSockPair(name string) (parent, child *os.File, err error) {
 	return os.NewFile(uintptr(fds[1]), name+"-p"), os.NewFile(uintptr(fds[0]), name+"-c"), nil
 }
 
-// WithProcfd runs the passed closure with a procfd path (/proc/self/fd/...)
-// corresponding to the unsafePath resolved within the root. Before passing the
-// fd, this path is verified to have been inside the root -- so operating on it
-// through the passed fdpath should be safe. Do not access this path through
-// the original path strings, and do not attempt to use the pathname outside of
-// the passed closure (the file handle will be freed once the closure returns).
-func WithProcfd(root, unsafePath string, fn func(procfd string) error) error {
```

Comment:
Sorry -- this is also a breaking change, but I don’t think we need to mark it as deprecated since we no longer recommend using it.

Reply:
Dammit, it seems like some people do use this externally...

We really need to just move everything into `internal` in runc 1.5.0, having to worry about this with every refactor PR is just becoming a waste of time.